src/backend/expungeservice/record_summarizer.py

```diff
@@ -93,16 +93,29 @@ def get_label(charge: Charge):
         return eligible_charges_by_date
 
     @staticmethod
-    def _build_is_all_cases_feeless(charges):
-        return bool(
-            charges
-            and next(
-                filter(
-                    lambda charge: charge.expungement_result.charge_eligibility.status
-                    == ChargeEligibilityStatus.ELIGIBLE_NOW
-                    and charge.disposition.status != DispositionStatus.CONVICTED,
-                    charges,
-                ),
-                None,
-            )
+    def _build_no_fees_reason(charges):
+        has_eligible_nonconvictions = charges and next(
+            filter(
+                lambda charge: charge.expungement_result.charge_eligibility.status
+                == ChargeEligibilityStatus.ELIGIBLE_NOW
+                and charge.disposition.status != DispositionStatus.CONVICTED,
+                charges,
+            ),
+            None,
         )
+        has_future_eligible_convictions = charges and next(
+            filter(
+                lambda charge: charge.expungement_result.charge_eligibility.status
+                == ChargeEligibilityStatus.WILL_BE_ELIGIBLE,
+                charges,
+            ),
+            None,
+        )
+        if has_future_eligible_convictions:
+            reason = "no convictions eligible now"
+        else:
+            if has_eligible_nonconvictions:
+                reason = "no eligible convictions"
+            else:
+                reason = "no eligible cases"
+        return reason
```

Comment:
I like the switch to reason strings; it makes it much clearer at glance what this will be used for in the front-end compared to `is_all_cases_feeless`.

In general, [list comp should be used over filter](https://www.artima.com/weblogs/viewpost.jsp?thread=98196) and I think that applies here too (especially given the rest of the codebase using list comps). The following is not tested.
```python3
reason = "no eligible cases" # reason only applies if feeless
if charges:
    charges_will_be_eligible = [c for c in charges if c.expungement_result.charge_eligibility.status == ChargeEligibilityStatus.WILL_BE_ELIGIBLE]
    nonconvictions_eligible_now = [c for c in charges if c.expungement_result.charge_eligibility.status == ChargeEligibilityStatus.ELIGIBLE_NOW and c.disposition.status != DispositionStatus.CONVICTED]
    if charges_will_be_eligible:
        reason = "no convictions eligible now"
    elif nonconvictions_eligible_now:
        reason = "no eligible convictions"
return reason
```